### PR TITLE
test: Introduce scaled timeouts 

### PR DESCRIPTION
**Pull request by xdustinface**
This PR introduces `--timeoutscale=<value>` in the test framework. This allows to scale the test timeouts by multiplying them with `<value>`. This is mostly meant to be used by CI where time seems to be rare from time to time.

### CODE REVIEW

**Comment by UdjinM6** [comment]
Few suggestions, pls see 8cd990e0e

**Comment by xdustinface** [comment]
The suggested `set_timeout_scale` thing doesn't work as you expect. It only changes the timeout for the util module imported by `test_framework`. In python this is obviously not a thing like static variables in c/c++. Just print `timeout_scale` in `wait_until` and run some tests.. it should always be 1.  I wanted to do the same initially but i as i noticed that this doesn't work and as i couldn't find a better way without a lot changes in the other files i did the thing with the parsing in util as workaround.

**Comment by UdjinM6** [comment]
Ahhh, right... 🙈  👍

**Comment by xdustinface** [comment]
With the force push i applied your other suggestion and squashed my linter fix commit.

**Comment by UdjinM6** [comment]
Still don't like parsing args in utils cause it feels like a circular dependency, pls see 1eaf89a483

**Comment by xdustinface** [comment]
Cool, thanks, thats what i was looking for 🙂

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import os
 import random
 import shutil
+import sys
```
Comment (UdjinM6) [inline test/functional/test_framework/util.py:18]:
```suggestion
```

**Comment by xdustinface** [comment]
Fixed and squashed!

**Pull request merged**